node/node.go

```diff
@@ -278,56 +278,69 @@ func (node *Node) tryBroadcastStaking(stakingTx *staking.StakingTransaction) {
 }
 
 // Add new transactions to the pending transaction list.
-func (node *Node) addPendingTransactions(newTxs types.Transactions) {
+func (node *Node) addPendingTransactions(newTxs types.Transactions) []error {
 	poolTxs := types.PoolTransactions{}
 	for _, tx := range newTxs {
 		poolTxs = append(poolTxs, tx)
 	}
-	node.TxPool.AddRemotes(poolTxs)
+	errs := node.TxPool.AddRemotes(poolTxs)
 
 	pendingCount, queueCount := node.TxPool.Stats()
 	utils.Logger().Info().
 		Int("length of newTxs", len(newTxs)).
 		Int("totalPending", pendingCount).
 		Int("totalQueued", queueCount).
 		Msg("Got more transactions")
+	return errs
 }
 
 // Add new staking transactions to the pending staking transaction list.
-func (node *Node) addPendingStakingTransactions(newStakingTxs staking.StakingTransactions) {
+func (node *Node) addPendingStakingTransactions(newStakingTxs staking.StakingTransactions) []error {
 	if node.NodeConfig.ShardID == shard.BeaconChainShardID &&
 		node.Blockchain().Config().IsPreStaking(node.Blockchain().CurrentHeader().Epoch()) {
 		poolTxs := types.PoolTransactions{}
 		for _, tx := range newStakingTxs {
 			poolTxs = append(poolTxs, tx)
 		}
-		node.TxPool.AddRemotes(poolTxs)
+		errs := node.TxPool.AddRemotes(poolTxs)
 		pendingCount, queueCount := node.TxPool.Stats()
 		utils.Logger().Info().
 			Int("length of newStakingTxs", len(poolTxs)).
 			Int("totalPending", pendingCount).
 			Int("totalQueued", queueCount).
 			Msg("Got more staking transactions")
+		return errs
 	}
+	return make([]error, len(newStakingTxs))
 }
 
 // AddPendingStakingTransaction staking transactions
 func (node *Node) AddPendingStakingTransaction(newStakingTx *staking.StakingTransaction) {
 	if node.NodeConfig.ShardID == shard.BeaconChainShardID {
-		node.addPendingStakingTransactions(staking.StakingTransactions{newStakingTx})
+		errs := node.addPendingStakingTransactions(staking.StakingTransactions{newStakingTx})
+		for i := range errs {
+			if errs[i] != nil {
+				return
```

Comment:
if there is at least one error, we don't broadcast any txns?

Reply:
ah nevermind. was confused because was a slice of errors, but its just one transaction.
Better to use if errs[0] != nil { return }